service/src/LevelZero.cpp

```diff
@@ -702,167 +737,203 @@ namespace geopm
                         "LevelZero::" + std::string(__func__) +
                         ": LevelZero Event Pool Create failed",
                         __LINE__);
+
+        // TODO: nothing guarantees CHIP 0 was called first
         m_devices.at(l0_device_idx).subdevice.event_pool.push_back(event_pool_handle);
 
         ze_event_desc_t event_desc = {ZE_STRUCTURE_TYPE_EVENT_DESC, nullptr, 0,
                                       ZE_EVENT_SCOPE_FLAG_HOST, ZE_EVENT_SCOPE_FLAG_HOST};
 
-        ze_event_handle_t event = nullptr;
-        ze_result = zeEventCreate(event_pool_handle, &event_desc, &event);
+        ze_event_handle_t metric_notifcation_event = nullptr;
+        ze_result = zeEventCreate(event_pool_handle, &event_desc, &metric_notifcation_event);
         check_ze_result(ze_result, GEOPM_ERROR_RUNTIME,
                         "LevelZero::" + std::string(__func__) +
                         ": LevelZero Event Create failed",
                         __LINE__);
 
-        m_devices.at(l0_device_idx).subdevice.event.push_back(event);
+        // TODO: nothing guarantees CHIP 0 was called first
+        m_devices.at(l0_device_idx).subdevice.metric_notifcation_event.push_back(metric_notifcation_event);
 
         zet_metric_streamer_desc_t metric_streamer_desc = {
             ZET_STRUCTURE_TYPE_METRIC_STREAMER_DESC,
             nullptr,
-            32768,
-            m_devices.at(l0_device_idx).metric_sampling_period};
+            4, // number of reports to notify on.
```

Comment:
Same question as before: why 4 reports at 1 ms per sample? Is that something to do with our typical 5ms agent loops?